### ISSUE TITLE
undefined variable when include parameter is a playbook to be included

### ISSUE THREAD

**Issue by etcmorph** [ansible/ansible#22360]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
parameterized playbook include

##### ANSIBLE VERSION

```
ansible 2.2.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION
n/a

##### OS / ENVIRONMENT
RedHat EL7

##### SUMMARY
Including a parameter with a playbook include works, except if that parameter is the name of a playbook to be included.

##### STEPS TO REPRODUCE
```
ansible-playbook include_include_play.yml
```

include_include_play.yml
```yaml
- include: include_play.yml play_yml=play.yml var=value
```
include_play.yml
```yaml
- include: "{{play_yml}}"
```
play.yml
```yaml
- name: test play
  hosts: localhost
  tasks:
    - debug: msg="{{var}}"
```

##### EXPECTED RESULTS

```
PLAY [test play] ***************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [debug] *******************************************************************
ok: [localhost] => {
    "msg": "value"
}
```
##### ACTUAL RESULTS
```
ERROR! 'play_yml' is undefined
```

If I run the same command with an extra-vars of "play_yml=play.yml" I get the expected output above.

**Comment by krzysztof-magosa** [comment]
Try:
```
- include: include_play.yml
  vars:
    play_yml: play.yml
    var: value
```

**Comment by etcmorph** [comment]
I did try that as well - forgot to mention sorry, same issue.

**Comment by bcoca** [comment]
Possible Misunderstanding
=========================

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us.

We believe the ticket you have filed is being somewhat misunderstood, as one thing works a little differently than stated. Only extra vars are available for templating on play includes.

In the future, this might be a topic more well suited for the user list, which you can also post here if you'd like some more help with the above.
   
   * https://groups.google.com/forum/#!forum/ansible-project

Thank you once again for this and your interest in Ansible!

**Comment by etcmorph** [comment]
Hi - thanks for the fast response!

> Only extra vars are available for templating on play includes.

It seems like a curious edge case to me why only extra-vars would work.
I would have expected either both to work or neither, which is why I filed
as a bug.

The closest I found in the docs was this:

[‘play’ includes are very limited and most directives do not work](http://docs.ansible.com/ansible/playbooks_roles.html#task-versus-play-includes)

Is it reasonable to request this as a feature?

Thanks,
